app/routes/login.tsx

```diff
@@ -7,21 +7,27 @@
  */
 
 import { Button } from '@oxide/design-system'
-import { json, type LoaderFunction } from '@remix-run/node'
-import { Form } from '@remix-run/react'
+import {
+  json,
+  redirect,
+  type ActionFunctionArgs,
+  type LoaderFunction,
+} from '@remix-run/node'
+import { Form, useActionData, useLoaderData } from '@remix-run/react'
+import { useState } from 'react'
 
-import { isAuthenticated } from '~/services/authn.server'
+import { auth, getUserFromSession } from '~/services/auth.server'
 import { returnToCookie } from '~/services/cookies.server'
-import { getUserRedirect } from '~/services/redirect.server'
 
 export let loader: LoaderFunction = async ({ request }) => {
   const url = new URL(request.url)
   const returnTo = url.searchParams.get('returnTo')
+  const emailResponse = url.searchParams.get('email')
 
-  // if we're already logged in, go straight to returnTo
-  await isAuthenticated(request, {
-    successRedirect: getUserRedirect(returnTo),
-  })
+  // If we're already logged in, go straight to returnTo
+  if (await getUserFromSession(request)) {
+    throw redirect('/')
+  }
```

Comment:
should this be going to the `returnTo` or should the comment be updated? I don't see anything else in the code that ever actually returns to the `returnTo`. I'm going to test it on the preview